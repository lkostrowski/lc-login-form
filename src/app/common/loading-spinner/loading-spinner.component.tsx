import cx from 'classnames';
import React, { SVGAttributes } from 'react';

import { ReactComponent as SpinnerIcon } from '../../../icons/loading.svg';

import styles from './loading-spinner.module.scss';

export const LoadingSpinner = ({
    className,
    ...props
}: SVGAttributes<SVGElement>) => {
    return <SpinnerIcon className={cx(className, styles.loader)} {...props} />;
};
