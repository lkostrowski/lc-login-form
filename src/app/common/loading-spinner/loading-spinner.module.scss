.loader {
    animation: spin 3s linear;
}

@keyframes spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
