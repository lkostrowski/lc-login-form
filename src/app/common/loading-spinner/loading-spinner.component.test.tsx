import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { LoadingSpinner } from './loading-spinner.component';

afterEach(cleanup);

describe('Loading spinner component', () => {
    describe('Snapshots', () => {
        test('Default', () => {
            const { container } = render(<LoadingSpinner />);

            expect(container).toMatchSnapshot();
        });
    });
});
