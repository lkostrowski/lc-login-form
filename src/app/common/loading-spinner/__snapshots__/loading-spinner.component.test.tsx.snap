// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading spinner component Snapshots Default 1`] = `
<div>
  <svg
    class="loader"
  >
    loading.svg
  </svg>
</div>
`;
